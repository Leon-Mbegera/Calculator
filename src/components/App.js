import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const data = Calculate(this.state, buttonName);
    this.setState({
      next: data.next,
      total: data.total,
      operation: data.operation,
    });
  }

  render() {
    return (
      <>
        <Display show={this.state.total || this.state.next} />
        <ButtonPanel clickHandler={() => handleClick(buttonName)} />
      </>
    );
  }
}

export default App;
