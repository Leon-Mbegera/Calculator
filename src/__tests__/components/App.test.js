import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../../components/App';

describe('App component', () => {
  test('contains a text element', () => {
    render(<App />);
    const text = screen.getByText('Lets do some Math!');
    expect(text).toBeInTheDocument();
  });
});
