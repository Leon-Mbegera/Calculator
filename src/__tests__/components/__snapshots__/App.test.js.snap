// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders Display and ButtonPanel components 1`] = `
<div
  className="contain"
>
  <h1
    className="heading"
  >
    Lets do some Math!
  </h1>
  <div
    className="border"
  >
    <div
      className="display-screen"
    >
      0
    </div>
    <div
      className="button-panel"
    >
      <div
        className="panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "lavender",
              "width": "25%",
            }
          }
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "lavender",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "lavender",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "lavender",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "whitesmoke",
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "background": "lavender",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
